numpy/core/src/multiarray/flagsobject.c

```diff
@@ -329,8 +337,13 @@ arrayflags_writeable_set(
                 "Cannot set flags on array scalars.");
         return -1;
     }
+
+    int istrue = PyObject_IsTrue(obj);
+    if (istrue == -1) {
+        return -1;
+    }
     res = PyObject_CallMethod(self->arr, "setflags", "OOO",
```

Comment:
Just a general comment.  `PyArray_CallMethodObjArgs` is a much nicer way to write this.  (no need to do, though)